src/libLookup/Lookup.cpp

```diff
@@ -5436,145 +5441,195 @@ void Lookup::RectifyTxnShardMap(const uint32_t oldNumShards,
   LOG_GENERAL(INFO, "Elapsed time for exchange " << elaspedTimeMs);
 }
 
-void Lookup::SendTxnPacketToNodes(const uint32_t oldNumShards,
-                                  const uint32_t newNumShards) {
+void Lookup::SendTxnPacketToShard(const uint32_t shardId, bool toDS) {
   LOG_MARKER();
 
   if (!LOOKUP_NODE_MODE) {
     LOG_GENERAL(WARNING,
-                "Lookup::SendTxnPacketToNodes not expected to be called from "
+                "Lookup::SendTxnPacketToShard not expected to be called from"
                 "other than the LookUp node.");
     return;
   }
 
-  if (m_mediator.m_disableTxns) {
-    LOG_GENERAL(INFO, "Txns disabled - skipping dispatch to shards");
-    return;
-  }
+  bytes msg = {MessageType::NODE, NodeInstructionType::FORWARDTXNPACKET};
+  bool result = false;
 
-  const uint32_t numShards = newNumShards;
+  {
+    unique_lock<mutex> g1(m_txnShardMapMutex, defer_lock);
+    unique_lock<mutex> g2(m_txnShardMapGeneratedMutex, defer_lock);
+    lock(g1, g2);
+
+    if (shardId > m_txnShardMapGenerated.size()) {
```

Comment:
when this is being called from `SendTxnPacketToDS`, map `m_txnShardMapGenerated` is cleared in `SendTxnPacketPrepare` and populated only for given shard id ( say 3 for ds ).
So above check will always fail and no txns are sent to DS. For ex: shard id 3 and `m_txnShardMapGenerated.size() = 1`
May be should better check for existing of key `shardid` in map `m_txnShardMapGenerated` instead?